FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/Jobs/SchedulerService.cs

```diff
@@ -472,26 +472,29 @@ private async Task<CurrentTriggerEntity> CreateInitialTriggerEntity(Cancellation
         /// So the nextTriggerEndTime is set JobEndTimeLatencyInMinutes earlier than the current timestamp.
         /// If Job end time is specified and earlier than the nextTriggerEndTime, will set the nextTriggerEndTime to the specified job end time.
         /// For initial load job, lastTriggerEndTime is null;
-        /// For incremental job, lastTriggerEndTime is the end time of the last trigger, we also check the next occurrence time.
-        /// If next occurrence time comes after nextTriggerEndTime, will return null and skip this iteration.
+        /// For incremental job, lastTriggerEndTime is the end time of the last trigger, we also get all the next occurrence times during lastTriggerEndTime and latency utc now.
+        /// If next occurrence time comes after latency utc now, will return null and skip this iteration.
         /// </summary>
         /// <param name="lastTriggerEndTime">the end time of the last trigger</param>
         /// <returns>the end time of the next trigger, return null if it is not yet the next occurrence time</returns>
         private DateTimeOffset? GetNextTriggerEndTime(DateTimeOffset? lastTriggerEndTime)
         {
+            // Add latency to utc now
             DateTimeOffset nextTriggerEndTime =
                 DateTimeOffset.UtcNow.AddMinutes(-1 * JobConfigurationConstants.JobQueryLatencyInMinutes);
 
+            // set the trigger end time for incremental job
             if (lastTriggerEndTime != null)
             {
-                DateTime nextOccurrenceTime =
-                    _crontabSchedule.GetNextOccurrence(((DateTimeOffset)lastTriggerEndTime).DateTime);
-                DateTimeOffset nextOccurrenceOffsetTime = DateTime.SpecifyKind(nextOccurrenceTime, DateTimeKind.Utc);
-                if (nextOccurrenceOffsetTime > nextTriggerEndTime)
+                // this functions will return times > baseTime and < endTime
+                List<DateTime> nextOccurrenceTimes = _crontabSchedule.GetNextOccurrences(((DateTimeOffset)lastTriggerEndTime).DateTime, nextTriggerEndTime.DateTime).ToList();
```

Comment:
Wondering if we can remove the ToList here because we only want the last element, the internal implementation is yield and does not compose a potential long list

Reply:
Remove ToList, and use LastOrDefault().